harvester/create.go

```diff
@@ -238,3 +168,85 @@ func (d *Driver) createKeyPair() error {
 	d.SSHPublicKey = string(publicKey)
 	return nil
 }
+
+func (d *Driver) addDisk(vmBuilder *builder.VMBuilder, disk *Disk, diskIndex int) (*builder.VMBuilder, error) {
+	diskName := fmt.Sprintf("%s-%d", diskNamePrefix, diskIndex)
+	if disk.Bus == "" {
+		disk.Bus = defaultDiskBus
+	}
+	if disk.Type == "" {
+		disk.Type = builder.DiskTypeDisk
+	}
+	isCDRom := disk.Type == builder.DiskTypeCDRom
+	var imageID string
+	if disk.ImageName != "" {
+		imageNamespace, imageName, err := NamespacedNamePartsByDefault(disk.ImageName, d.VMNamespace)
+		if err != nil {
+			return nil, err
+		}
+		imageID = fmt.Sprintf("%s/%s", imageNamespace, imageName)
+		disk.StorageClassName = builder.BuildImageStorageClassName("", imageName)
+	}
+	pvcOption := &builder.PersistentVolumeClaimOption{
+		ImageID:          imageID,
+		StorageClassName: pointer.StringPtr(disk.StorageClassName),
+		VolumeMode:       corev1.PersistentVolumeBlock,
+		AccessMode:       corev1.ReadWriteMany,
+	}
+	return vmBuilder.PVCDisk(diskName, disk.Bus, isCDRom, disk.HotPlugAble, disk.BootOrder, fmt.Sprintf("%dGi", disk.Size), "", pvcOption), nil
```

Comment:
Need to handle bootOrder equal to 0 cases after JSON format as the minimum boot order should be >= 1.

Reply:
handled in builder